diff --git a/dist/cjs/dev-tools/dev-middleware/hmr-client/createSocketUrl.js b/dist/cjs/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
index 753292b7c627f353132050ca7c7e0738c9825150..c8b6b21f4ad5b68e224b3cbe5e96f21ae9212104 100644
--- a/dist/cjs/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
+++ b/dist/cjs/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
@@ -34,7 +34,7 @@ function createSocketUrl(resourceQuery) {
   return getSocketUrl(options, currentLocation);
 }
 function formatURL({ port, protocol, hostname, pathname }) {
-  if (window.URL) {
+  if (typeof URL !== "undefined") {
     const url = new URL("http://localhost");
     url.port = port;
     url.hostname = hostname;
diff --git a/dist/esm/dev-tools/dev-middleware/hmr-client/createSocketUrl.js b/dist/esm/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
index c63139a166c2c687cba52c41d438da0eb754b6c2..9c9a1465d69848f394e620a304f884c1e59d291d 100644
--- a/dist/esm/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
+++ b/dist/esm/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
@@ -28,7 +28,7 @@ function createSocketUrl(resourceQuery) {
 }
 function formatURL(param) {
   var port = param.port, protocol = param.protocol, hostname = param.hostname, pathname = param.pathname;
-  if (window.URL) {
+  if (typeof URL !== "undefined") {
     var url = new URL("http://localhost");
     url.port = port;
     url.hostname = hostname;
diff --git a/dist/esm-node/dev-tools/dev-middleware/hmr-client/createSocketUrl.js b/dist/esm-node/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
index b0c6f621af82207e846f3cce20b6435b3389e8db..1c10c721011296fd3ef58ef3a293fbde53c51164 100644
--- a/dist/esm-node/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
+++ b/dist/esm-node/dev-tools/dev-middleware/hmr-client/createSocketUrl.js
@@ -10,7 +10,7 @@ function createSocketUrl(resourceQuery) {
   return getSocketUrl(options, currentLocation);
 }
 function formatURL({ port, protocol, hostname, pathname }) {
-  if (window.URL) {
+  if (typeof URL !== "undefined") {
     const url = new URL("http://localhost");
     url.port = port;
     url.hostname = hostname;
