diff --git a/dist/import.mjs b/dist/import.mjs
index d86edf1f194e0495ed4e11f3814f52efbaefe8c7..62f891f3b81d7fd47950c2100ade3700d39700e3 100644
--- a/dist/import.mjs
+++ b/dist/import.mjs
@@ -1396,14 +1396,15 @@ function $e0b6e0b68ec7f50f$export$872b660ac5a1ff98(props) {
 }
 function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
     if (event.button > 0) return false;
+    const eventTarget = event.composedPath()[0];
     if (event.target) {
         // if the event target is no longer in the document, ignore
         const ownerDocument = event.target.ownerDocument;
         if (!ownerDocument || !ownerDocument.documentElement.contains(event.target)) return false;
         // If the target is within a top layer element (e.g. toasts), ignore.
-        if (event.target.closest("[data-react-aria-top-layer]")) return false;
+        if (eventTarget.closest("[data-react-aria-top-layer]")) return false;
     }
-    return ref.current && !ref.current.contains(event.target);
+    return ref.current && !ref.current.contains(eventTarget);
 }
 
 
