diff --git a/dist/useInteractOutside.mjs b/dist/useInteractOutside.mjs
index 294e75e5117431f357cbfe57a87233f643b30686..a2794471d2a364fdb4d0f657570a388bc319b138 100644
--- a/dist/useInteractOutside.mjs
+++ b/dist/useInteractOutside.mjs
@@ -81,14 +81,16 @@ function $e0b6e0b68ec7f50f$export$872b660ac5a1ff98(props) {
 }
 function $e0b6e0b68ec7f50f$var$isValidEvent(event, ref) {
     if (event.button > 0) return false;
+    // compat for ShadowDOM
+    const eventTarget = event.composedPath()[0];
     if (event.target) {
         // if the event target is no longer in the document, ignore
         const ownerDocument = event.target.ownerDocument;
         if (!ownerDocument || !ownerDocument.documentElement.contains(event.target)) return false;
         // If the target is within a top layer element (e.g. toasts), ignore.
-        if (event.target.closest('[data-react-aria-top-layer]')) return false;
+        if (eventTarget.closest("[data-react-aria-top-layer]")) return false;
     }
-    return ref.current && !ref.current.contains(event.target);
+    return ref.current && !ref.current.contains(eventTarget);
 }
 
 
